plugins/11_TheWarWithin/zones/undermine.lua

```diff
@@ -0,0 +1,552 @@
+        Pet({item = 232846, id = 4648, count = 5}), -- Steamwheedle Flunkie
+        Pet({item = 232849, id = 4650, count = 5}), -- venture Companyman
+        Pet({item = 232840, id = 4661, count = 5}), -- Mechagopher
+        Pet({item = 232841, id = 4644, count = 8}), -- Professor Punch
+        Pet({item = 232842, id = 4638, count = 10}) -- Crimson Mechasaur
+    }
+}) -- Ditty Fuzeboy
+
+map.nodes[25743813] = Vendor({
+    id = 234776,
+    note = L['angelo_rustbin_note'],
+    rewards = {
+        Pet({item = 232858, id = 4655, count = 1}), -- Cruncher
+        Pet({item = 232859, id = 4653, count = 3}), -- Lab Rat
+        Item({item = 232986, count = 1}) -- GE86 Advance
+    }
+}) -- Angelo Rustbin
+
+-------------------------------------------------------------------------------
+----------------------------- S.C.R.A.P. REWARDS ------------------------------
+-------------------------------------------------------------------------------
+
+local function GetScrapRewards(criteriaID)
+    return {
+        Achievement({id = 41593, criteria = criteriaID}), -- Cleanin' the Streets
+        Achievement({
+            id = 41592,
+            criteria = {id = 1, qty = true, suffix = L['scrap_heap_suffix']}
+        }), -- Absolutely Zero Littering
+        Achievement({id = 41594, oneline = true})
+    }
+end
+
+local ScrapHeap = Class('ScrapHeap', Collectible, {
+    label = '{npc:234350}',
+    icon = 5768266,
+    group = ns.groups.SCRAP_REWARDS
+}) -- S.C.R.A.P. Heap
+
+function ScrapHeap.getters:rewards() return GetScrapRewards(self.criteriaID) end
+
+map.nodes[32012173] = ScrapHeap({criteriaID = 103181}) -- Hovel Hill (North)
+map.nodes[36354508] = ScrapHeap({criteriaID = 103182}) -- Hovel Hill (South)
+map.nodes[38658109] = ScrapHeap({criteriaID = 103183}) -- The Vatworks
+map.nodes[50426363] = ScrapHeap({criteriaID = 103184}) -- Venturewood
+map.nodes[52498333] = ScrapHeap({criteriaID = 103185}) -- The Heaps
+map.nodes[69577652] = ScrapHeap({criteriaID = 103186}) -- Emerald Hills Golf Course
+map.nodes[67482929] = ScrapHeap({criteriaID = 103187}) -- Blackwater Marina
+
+hooksecurefunc(VignettePinMixin, 'OnMouseEnter', function(self)
+    if self.vignetteID ~= 6687 and self.vignetteID ~= 6757 then return end
+    if not ns.groups.SCRAP_REWARDS:IsEnabled() then return end
+    local mapID = self:GetMap().mapID
+    local x = C_VignetteInfo.GetVignettePosition(self.vignetteGUID, mapID).x
+    local y = C_VignetteInfo.GetVignettePosition(self.vignetteGUID, mapID).y
+    local coordinates = HandyNotes:getCoord(x, y)
+    local node = map.nodes[coordinates]
+    if not node then return end
+    if ns:GetOpt('show_loot') then
+        local rewards = GetScrapRewards(node.criteriaID)
+        for i, reward in ipairs(rewards) do
+            if reward:IsEnabled() then reward:Render(GameTooltip) end
+        end
+        GameTooltip:AddLine(' ')
+    end
+    GameTooltip:Show()
+end)
```

Comment:
We've done this type of thing a few times. I wonder if we can generalize into some sort of utility function.

Reply:
@zarillion I started this process by moving the existing code to `tooltip.lua`. We would need to do an audit of all instances where we utilize `hooksecurefunc(VignettePinMixin ...` to see how much work it would take to build a more robust function but for now it's started!